google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -213,13 +214,18 @@ public B setHost(String host) {
     }
 
     /**
-     * Sets the service authentication credentials.
+     * Sets the service authentication credentials. If this method or {@link #setNoCredentials() are
+     * not used on the builder, {@link GoogleCredentials#getApplicationDefault()} will be used to
+     * attempt getting credentials from the environment.
      *
+     * @param credentials authentication credentials, should not be {@code null}
      * @return the builder
+     * @throws NullPointerException if {@code credentials} is {@code null}. To disable
+     *     authentication use {@link Builder#setNoCredentials()}
```

Comment:
I would prefer not to add types other than the ones in `google-auth-library-java`. Users already struggle with all the different credentials we have and it makes little sense to adopt a library for authentication and then start adding custom types.

We could consider adding a `NullCredentials` to `google-cloud-java` (or `NoAuthCredentials`) to get rid of the `setNoCredentials` method for supporting the emulator case, this seems as a reasonable compromise but I am strongly against adding `DefaultCredentials` class.